kite-data/kite-data-core/src/main/java/org/kitesdk/data/View.java

```diff
@@ -128,4 +128,28 @@
    * @since 0.16.0
    */
   public URI getUri();
+
+  /**
+   * Returns an {@code Iterable} of non-overlapping {@link View} objects that
+   * partition the underlying {@link Dataset} and cover this {@code View}.
+   *
+   * The returned {@code View} objects are implementation-specific, but should
+   * represent reasonable partitions of the underlying {@code Dataset} based on
+   * its layout.
+   *
+   * The data contained by the union of each {@code View} in the
+   * {@code Iterable} must be a super-set of this {@code View}.
+   *
+   * Note that partitions are actual partitions under which data is stored.
+   * Implementations should omit any {@code View} that is empty.
+   *
+   * @return
+   *      An Iterable of the {@code View} that cover this {@code View}.
+   * @throws IllegalStateException
+   *      If the underlying {@code Dataset} is not partitioned.
+   * @throws UnsupportedOperationException
+   *      If the implementation does not support covering partitions.
+   * @since 1.1.0
+   */
+  public Iterable<PartitionView<E>> getCoveringPartitions();
```

Comment:
Understanding that we want to grant flexibility to the implementations, but I'm wondering if we can strengthen the guarantees of the returned partitions a bit. For instance, if I have a year/month/day partition, getting the covering partitions for view:...example?year=2015&month=5 could do any of the following per this specification:

* A single-element iterable for the View containing the entire year of 2015 
* A single-element iterable view for just May of 2015 
* A 31-element iterable for each day in May of 2015

At the very least I think we should guarantee getCoveringPartitions() will return no views that are disjoint from itself. For instance, getCoveringPartitions() for year=2015&month=5 won't return the year=2015&month=6 partition, since that is disjoint from the partition we're invoking this on. (This seems obvious but isn't disallowed according to this spec.)

Perhaps we could guarantee that getCoveringPartition returns the minimal number of non-overlapping partitions that contain some data in the given view? This should make it easier to write tooling in the future to, say, efficiently archive off old data.